tlf/handle_extension.go

```diff
@@ -123,16 +123,28 @@ type HandleExtension struct {
 
 // String implements the fmt.Stringer interface for HandleExtension.
 // Ex: "(conflicted copy 2016-05-09 #2)"
-func (e HandleExtension) String() string {
+func (e HandleExtension) string(isBackedByTeam bool) string {
 	date := time.Unix(e.Date, 0).UTC().Format(handleExtensionDateFormat)
 	var num string
-	if e.Number > 1 {
+	minNumberSuffixToShow := uint16(2)
+	if isBackedByTeam {
+		// When a TLF is backed by an implicit team, it should always
+		// use the "#1" suffix, unlike for older TLFs.
+		minNumberSuffixToShow = 1
+	}
+	if e.Number >= minNumberSuffixToShow {
 		num = " #"
 		num += strconv.FormatUint(uint64(e.Number), 10)
 	}
 	return fmt.Sprintf(handleExtensionFormat, e.Type.String(e.Username), date, num)
 }
 
+// String implements the fmt.Stringer interface for HandleExtension.
+// Ex: "(conflicted copy 2016-05-09 #2)"
+func (e HandleExtension) String() string {
+	return e.string(false)
```

Comment:
I wonder if this should be the default. Can we get rid of this interface implementation?

Reply:
Sorry, I don't follow the suggestion.  You want to get rid of `String()` entirely?  I'd rather not do that, since it'll break any log message that prints the extension directly, and we won't know that it's broken.

Or do you want to flip this to `e.string(true)`?  That might make sense once we turn on implicit teams on the server, but even then there will probably be a long time when the majority of the folders are not team-backed, at least until migration happens.

Reply:
Alternatively: can we store the `isBackedByTeam` state in the object itself? It still won't address when it's loaded from the database, but at least when it's created on the fly it should show the right thing.

Reply:
I looked into this a bit -- there are a few places where we parse a string to create the handle extension, before we know anything about the backing TLF ID (e.g., when first parsing a folder name passed in on the command line).  So there's no way to know if it's back by a team at that point.

In general, whether to show the "#1" or not really only matters when you're comparing the string against something else for an equality check, which is what happens in the chat inbox.  Whether we show it in logs, or in the folder list or whatever, really doesn't matter.  So in my view, the default doesn't matter, I'm only trying to fix compatibility issues with the chat inbox.

Reply:
All right, sounds good.